lms/djangoapps/survey/models.py

```diff
@@ -219,6 +228,12 @@ def save_answers(cls, form, user, answers):
             # See if there is an answer stored for this user, form, field_name pair or not
             # this will allow for update cases. This does include an additional lookup,
             # but write operations will be relatively infrequent
-            answer, __ = SurveyAnswer.objects.get_or_create(user=user, form=form, field_name=name)
+            answer, __ = SurveyAnswer.objects.get_or_create(
+                user=user,
+                form=form,
+                field_name=name
+            )
+            if course_key:
+                answer.course_key = course_key
```

Comment:
using the defaults argument of get_or_create() will save us the additional save() query at write time. The fields specified in defaults are not used as a filter for get() but are used for object creation. Will create a separate PR to do this.
https://docs.djangoproject.com/en/1.8/ref/models/querysets/#get-or-create